src/editorial-source-components/TextInput.tsx

```diff
@@ -1,10 +1,32 @@
+import type { Theme } from "@emotion/react";
+import type { StyledComponent } from "@emotion/styled";
 import styled from "@emotion/styled";
+import type { SerializedStyles } from "@emotion/utils";
 import { space } from "@guardian/src-foundations";
 import { textSans } from "@guardian/src-foundations/typography";
+import type { Props } from "@guardian/src-helpers";
+import type { Width } from "@guardian/src-text-input";
 import { TextInput as OriginalTextInput } from "@guardian/src-text-input";
+import type { InputHTMLAttributes } from "react";
 import { inputBorder } from "./inputBorder";
 
-export const TextInput = styled(OriginalTextInput)`
+interface TextInputProps extends InputHTMLAttributes<HTMLInputElement>, Props {
```

Comment:
A bit unclear on how the changes in this file relate to the PR.

Reply:
That's fair, I'll add a comment. Because `TextInputProps` aren't directly exported by Source, we can't generate a definition file with them. Instead, we have to define them manually. It's a bit rubbish.